src/main/java/com/fincons/service/employeeService/impl/DepartmentServiceImpl.java

```diff
@@ -24,128 +23,109 @@ public class DepartmentServiceImpl implements DepartmentService {
-        ValidateSingleField.validateSingleField(departmentCode);
         Department department = validateDepartmentByCode(departmentCode);
         departmentRepository.deleteById(department.getId());
     }
 
     @Override
     public List<EmployeeDepartmentDTO> getDepartmentEmployeesFindByCodeDepartment(String departmentCode) {
-        ValidateSingleField.validateSingleField(departmentCode);
+        ValidateFields.validateSingleField(departmentCode);
         Department department = validateDepartmentByCode(departmentCode);
 
         return departmentRepository.getDepartmentEmployeesFindByIdDepartment(department.getId());
     }
 
-    public Department validateDepartmentByCode(String departmentCode){
+    public Department validateDepartmentByCode(String departmentCode) {
         Department existingDepartment = departmentRepository.findDepartmentByDepartmentCode(departmentCode);
 
-        if(Objects.isNull(existingDepartment)){
+        if (Objects.isNull(existingDepartment)) {
             throw new ResourceNotFoundException("Department with code: " + departmentCode + " not found");
         }
         return existingDepartment;
     }
 
-    public void validateDepartmentFields(DepartmentDTO departmentDTO){
+    public void validateDepartmentFields(DepartmentDTO departmentDTO) {
         //If one field is true run Exception
-        if (Strings.isEmpty(departmentDTO.getDepartmentCode())||
-                Strings.isEmpty(departmentDTO.getName())||
+        if (Strings.isEmpty(departmentDTO.getDepartmentCode()) ||
+                Strings.isEmpty(departmentDTO.getName()) ||
                 Strings.isEmpty(departmentDTO.getAddress()) ||
                 Strings.isEmpty(departmentDTO.getCity())) {
             throw new IllegalArgumentException("The fields of the Department can't be null or empty");
         }
     }
-    public void checkForDuplicateDepartment(DepartmentDTO departmentDTO, List<Department> departments){
-        for (Department department1 : departments) {
-            if (department1.getName().equals(departmentDTO.getName())) {
-                throw new DuplicateException("Department with the same name, already exists");
-            } else if (department1.getDepartmentCode().equals(departmentDTO.getDepartmentCode())) {
-                throw new DuplicateException("Department with the same code, already exists");
 
-            }
+    public void checkForDuplicateDepartment(String departmentCode, String departmentName) {
+
+        Department departmentByCode = departmentRepository.findDepartmentByDepartmentCode(departmentCode);
+        Department departmentByName = departmentRepository.findDepartmentByName(departmentName);
+        if (departmentByCode != null) {
+            throw new DuplicateException("Department with the same code, already exists");
+        }
+        if (departmentByName != null) {
+            throw new DuplicateException("Department with the same name, already exists");
```

Comment:
La classe DuplicateException è un estensione di una semplice RuntimeException, prova a migliorare la gestione delle eccezioni in questo modo: 
- fornisci un messaggio generico per l'eccezione nella classe DuplicateException :  "Impossibile inserire il nuovo $NomeOggetto nel database, il $NomeCampo risulta già esistente".
- lato metodo dove verrà lanciata poi dovrai passare solo le $variabili indicate nel messaggio.